app/celery/process_ses_receipts_tasks.py

```diff
@@ -342,3 +348,65 @@ def process_ses_smtp_results(
         current_app.logger.exception(e)
         current_app.logger.error('Error processing SES SMTP results: %s', type(e))
         self.retry(queue=QueueNames.RETRY)
+
+
+def check_and_queue_va_profile_email_status_callback(notification: Notification) -> None:
+    """
+    This checks the feature flag is enabled. If it is, queues the celery task and collects data from the notification.
+    Otherwise, it only logs a message.
+
+    :param notification: the email notification to collect data from
+    """
+
+    current_app.logger.debug(
+        'Sending email status to VA Profile, checking feature flag... | notification %s', notification.id
+    )
+
+    if is_feature_enabled(FeatureFlag.VA_PROFILE_EMAIL_STATUS_ENABLED):
+        current_app.logger.debug(
+            'Sending email status to VA Profile, collecting data for notification %s', notification.id
+        )
+        notification_data = {
+            'id': str(notification.id),  # this is the notification id
+            'reference': notification.client_reference,
+            'to': notification.to,  # this is the recipient's contact info (email)
+            'status': notification.status,  # this will specify the delivery status of the notification
+            'status_reason': notification.status_reason,  # populated if there's additional context on the delivery status
+            'created_at': notification.created_at.strftime(DATETIME_FORMAT),  # noqa: F821
+            'completed_at': notification.updated_at.strftime(DATETIME_FORMAT) if notification.updated_at else None,
+            'sent_at': notification.sent_at.strftime(DATETIME_FORMAT) if notification.sent_at else None,
+            'notification_type': notification.notification_type,  # this is the channel/type of notification (email)
+            'provider': notification.sent_by,  # email provider
+        }
+
+        # data passed to tasks must be JSON serializable
+        send_email_status_to_va_profile.delay(notification_data)
+    else:
+        current_app.logger.debug(
+            'Email status not sent to VA Profile, feature flag disabled | notification %s', notification.id
+        )
+
+
+@notify_celery.task(
+    throws=(AutoRetryException,),
+    autoretry_for=(AutoRetryException,),
+    max_retries=60,
+    retry_backoff=True,
+    retry_backoff_max=3600,
+)
+def send_email_status_to_va_profile(notification_data: dict) -> None:
```

Comment:
Nit and not a blocker:
Should `send_va_profile_email_status` and `send_email_status_to_va_profile` all be in one method. 
Right now it appears `send_email_status_to_va_profile` only responsibility is to call `send_va_profile_email_status` and raise an exception if it fails. 

Is it currently setup this way so AutoRetry functions properly?

Reply:
`send_email_status_to_va_profile` defines the celery task. I was originally collecting the info from the notification there too, but that had to get moved out to the calling function because info passed to a celery task has to be json serializable. I also don't want to define a celery task within the `VAProfileClient` class itself either. So this is what we are left with. Hopefully that makes sense.

Reply:
Sounds good to me!